### PR TITLE
fix: docker_swarm_service does not publish both tcp and udp ports

### PR DESCRIPTION
**Pull request by mit0223**
##### SUMMARY

fix: docker_swarm_service does not publish both tcp and udp ports for same published port

##### ISSUE TYPE

Bugfix Pull Request

##### COMPONENT NAME

docker_swarm_service

##### ADDITIONAL INFORMATION
this is fix for the issue
https://github.com/ansible/ansible/issues/60568

### CODE REVIEW

File: lib/ansible/modules/cloud/docker/docker_swarm_service.py [hunk 1]
```diff
@@ -2028,19 +2028,16 @@ def build_networks(self, docker_networks):
     def build_endpoint_spec(self):
         endpoint_spec_args = {}
         if self.publish is not None:
-            ports = {}
+            ports = []
             for port in self.publish:
+                port_spec = {
+                  'Protocol' : port['protocol'],
```
Comment (felixfontein) [inline lib/ansible/modules/cloud/docker/docker_swarm_service.py]:
```suggestion
                    'Protocol': port['protocol'],
```
You need to indent by four spaces, and remove the space before `:`. (Same for the next two lines.)

**Review by felixfontein — COMMENTED** [review]
This should fix the issue. You need to fix the linting problems (see my comment below), and you need to add a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs-how-to).

Something like
```.yaml
bugfixes:
- "docker_swarm_service - allow the same port to be published both with TCP and UDP."
```

Also, adding an integration test for this would be nice. The current tests for `publish` are [here](https://github.com/ansible/ansible/blob/devel/test/integration/targets/docker_swarm_service/tasks/tests/options.yml#L1486-L1613). You could add another task (`publish_8`) which publishes the same port with both protocols, then use `docker_swarm_service_info` to inspect the service, and assert that the resulting service has two published ports.

**Comment by mit0223** [comment]
Hi, felixfontein. I have handled your all comment.

**Review by felixfontein — APPROVED** [review]
LGTM. @jwitko @hannseman want to take a look as well?

**Review by jwitko — APPROVED** [review]
LGTM!

**Review by hannseman — APPROVED** [review]
shipit

**Comment by felixfontein** [comment]
@mit0223 thanks a lot for fixing this!
@jwitko @hannseman thanks for reviewing this!

I'll create a backport to stable-2.8 later.

**Pull request merged**